dowhy/causal_model.py

```diff
@@ -18,347 +19,378 @@
-            interpreter(self, **kwargs).interpret()
-
-    def summary(self, print_to_stdout=False):
-        """Print a text summary of the model.
-
-        :returns: a string containining the summary
-
-        """
-        summary_text = "Model to find the causal effect of treatment {0} on outcome {1}".format(self._treatment, self._outcome)
-        self.logger.info(summary_text)
-        if print_to_stdout:
-            print(summary_text)
-        return summary_text
+	"""Main class for storing the causal model state.
+
+	"""
+
+	def __init__(self, data, treatment, outcome, graph=None,
+				 common_causes=None, instruments=None,
+				 effect_modifiers=None,
+				 estimand_type="nonparametric-ate",
+				 proceed_when_unidentifiable=False,
+				 missing_nodes_as_confounders=False,
+				 **kwargs):
+		"""Initialize data and create a causal graph instance.
+
+		Assigns treatment and outcome variables.
+		Also checks and finds the common causes and instruments for treatment
+		and outcome.
+
+		At least one of graph, common_causes or instruments must be provided.
+
+		:param data: a pandas dataframe containing treatment, outcome and other
+		variables.
+		:param treatment: name of the treatment variable
+		:param outcome: name of the outcome variable
+		:param graph: path to DOT file containing a DAG or a string containing
+		a DAG specification in DOT format
+		:param common_causes: names of common causes of treatment and _outcome. Only used when graph is None.
+		:param instruments: names of instrumental variables for the effect of
+		treatment on outcome. Only used when graph is None.
+		:param effect_modifiers: names of variables that can modify the treatment effect. If not provided, then the causal graph is used to find the effect modifiers. Estimators will return multiple different estimates based on each value of effect_modifiers.
+		:param estimand_type: the type of estimand requested (currently only "nonparametric-ate" is supported). In the future, may support other specific parametric forms of identification.
+		:param proceed_when_unidentifiable: does the identification proceed by ignoring potential unobserved confounders. Binary flag.
+		:param missing_nodes_as_confounders: Binary flag indicating whether variables in the dataframe that are not included in the causal graph, should be  automatically included as confounder nodes.
+		:returns: an instance of CausalModel class
+
+		"""
+		self._data = data
+		self._treatment = parse_state(treatment)
+		self._outcome = parse_state(outcome)
+		self._effect_modifiers = parse_state(effect_modifiers)
+		self._estimand_type = estimand_type
+		self._proceed_when_unidentifiable = proceed_when_unidentifiable
+		self._missing_nodes_as_confounders = missing_nodes_as_confounders
+		self.logger = logging.getLogger(__name__)
+
+		if graph is None:
+			self.logger.warning("Causal Graph not provided. DoWhy will construct a graph based on data inputs.")
```

Comment:
you may want to shift  and copy this warning inside the if conditions? Because it is no longer true if the user has not provided any additional parameters and intends to learn it later.

Reply:
I added an extra comment inside the last else. Does that seem alright? I did not exactly understand what you meant by shifting inside the if conditions here.